### PR TITLE
nit: panic with error

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by chatton** [comment]
Doing a quick search I noticed we are relatively inconsistent with panicking on strings vs errors, maybe we could turn them all into errors?

**Comment by DimitrisJim** [comment]
For the instances @chatton mentions: `grep -inr "panic(\"" . --include="*.go` should find them

**Comment by chatton** [comment]
@DimitrisJim with next level grep-fu as usual :D

File: e2e/tests/upgrades/upgrade_test.go [hunk 1]
```diff
@@ -251,6 +251,201 @@ func (s *UpgradeTestSuite) TestChainUpgrade() {
 	})
 }
 
+// func (s *UpgradeTestSuite) TestV5ToV6ChainUpgrade() {
```
Comment (chatton) [inline e2e/tests/upgrades/upgrade_test.go]: did you accidentally pull intertx back in? 

intertx rising from the ashes!

File: modules/core/02-client/migrations/v7/solomachine.go [hunk 2]
```diff
@@ -52,99 +54,99 @@ func (cs ConsensusState) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error
 
 // ClientType panics!
 func (ClientState) ClientType() string {
-	panic("legacy solo machine is deprecated!")
+	panic(fmt.Errorf("legacy solo machine is deprecated!"))
```
Comment (DimitrisJim) [inline modules/core/02-client/migrations/v7/solomachine.go:57]: damn, revive won't even let us display some enthusiasm about a deprecation.

**Review by chatton — APPROVED** [review]
LGTM! I see 2 more in 

`callbacks/ibc_go_middleware_test.go`

**Comment by crodriguezvega** [comment]
> LGTM! I see 2 more in
> 
> `callbacks/ibc_go_middleware_test.go`

Yeah, I am going to leave those two. They are in tests, I don't think we need to be that consistent. :)

**Review by DimitrisJim — APPROVED** [review]

**Pull request merged**